packages/netsuite-adapter/src/filters/fix_order_problems_in_workbooks.ts

```diff
@@ -0,0 +1,199 @@
+  workbookTools
+    .filter(values.isPlainObject)
+    .flatMap(obj => Object.values(obj))
+    .forEach((workbookTool: Values) => {
+      if (_.isArray(workbookTool[DATASETS])) {
+        workbookTool[DATASETS].sort()
+      }
+      if (workbookTool[SOAP_SCRIPT_ID] in oldWorkbookToolsRecord) {
+        const oldWorkbookTool = oldWorkbookToolsRecord[workbookTool[SOAP_SCRIPT_ID]]
+        compareAndAssignInnerXmls(workbookTool, oldWorkbookTool)
+      }
+    })
+}
+
+const discardUnrelevantChanges = (
+  newInstanceValues: Values,
+  oldTools: Record<string, Values>,
+): void => {
+  INNER_ARRAY_NAMES.forEach((arrName: string) => {
+    if (Array.isArray(newInstanceValues[arrName])) {
+      adjustWorkbookToolsOrder(newInstanceValues[arrName], oldTools)
+    }
+  })
+}
+
+const addPermissionToDataset = (
+  dataset1Name: string,
+  dataset2Name: string,
+  datasetMap: Map<string, InstanceElement>,
+): void => {
+  const firstDataset = datasetMap.get(dataset1Name)
+  const secondDataset = datasetMap.get(dataset2Name)
+  if (
+    isInstanceElement(firstDataset)
+    && isInstanceElement(secondDataset)
+    && Array.isArray(secondDataset.value[DEPENDENCIES]?.[DEPENDENCY])
+  ) {
+    secondDataset.value[DEPENDENCIES]?.[DEPENDENCY]
+      ?.push(new ReferenceExpression(firstDataset.elemID, firstDataset.value[SCRIPT_ID], firstDataset))
+  }
+}
+
+const getDatasetLinkList = (workbook: InstanceElement): Values[] =>
+  (Array.isArray(workbook.value[DATASET_LINKS])
+    ? workbook.value[DATASET_LINKS]
+      .filter(values.isPlainObject)
+      .map((obj:Values) => obj[DATASET_LINK])
+      .filter(values.isPlainObject)
+    : [])
+
+const filterCreator: LocalFilterCreator = ({ elementsSource }) => ({
+  name: 'parseAnalytics',
+  onFetch: async elements => {
+    const workbooks = elements
+      .filter(elem => elem.elemID.typeName === WORKBOOK)
+      .filter(isInstanceElement)
+    await awu(workbooks)
+      .forEach(async instance => {
+        const oldTools = await createRecordOfPivotsChartsAndDsLinks(elementsSource, instance.elemID)
+        discardUnrelevantChanges(instance.value, oldTools)
+      })
+  },
+
+  preDeploy: async (changes: Change[]) => {
+    const changedDatasets = changes
+      .filter(isAdditionOrModificationChange)
+      .filter(change => change.data.after.elemID.typeName === DATASET)
+      .map(getChangeData)
+      .filter(isInstanceElement)
+
+    const datasetMap = new Map<string, InstanceElement>()
+    changedDatasets.forEach(dataset =>
+      datasetMap.set(
+        dataset.value[SCRIPT_ID],
+        dataset
+      ))
+
+    const workbooks = changes
+      .filter(isAdditionOrModificationChange)
+      .filter(change => change.data.after.elemID.typeName === WORKBOOK)
+      .map(getChangeData)
+      .filter(isInstanceElement)
+
+    workbooks.forEach(workbook => {
+      getDatasetLinkList(workbook)
+        .forEach((dslink: Values) => {
```

Comment:
dsLink is already `Values`
```suggestion
        .forEach(dslink => {
```